assets/tests/components/detours/diversionPage.delete.test.tsx

```diff
@@ -0,0 +1,133 @@
+      TestGroups.DetoursPilot,
+      TestGroups.DetoursList,
+      TestGroups.DeleteDraftDetours,
+    ])
+})
+
+describe("Detours Page: Open a Detour", () => {
+  test("calls API with correct detour ID", async () => {
+    jest
+      .mocked(fetchDetours)
+      .mockResolvedValue(Ok(detourListFactoryWithDraft.build()))
+
+    render(<DetourListPage />)
+
+    await userEvent.click(await screen.findByText("Draft Detour 123"))
+    expect(fetchDetour).toHaveBeenCalledWith(123)
+  })
+
+  test("renders detour details modal with delete draft button", async () => {
+    jest
+      .mocked(fetchDetours)
+      .mockResolvedValue(Ok(detourListFactoryWithDraft.build()))
+    jest.mocked(fetchDetour).mockResolvedValue(Ok(draftDetourFactory.build()))
+
+    render(<DetourListPage />)
+
+    await userEvent.click(await screen.findByText("Draft Detour 123"))
+
+    expect(deleteDetourButton.get()).toBeVisible()
+  })
+
+  test("delete draft detour confirmation modal displays when Delete Draft button clicked", async () => {
+    jest
+      .mocked(fetchDetours)
+      .mockResolvedValue(Ok(detourListFactoryWithDraft.build()))
+    jest.mocked(fetchDetour).mockResolvedValue(Ok(draftDetourFactory.build()))
+
+    render(<DetourListPage />)
+
+    await userEvent.click(await screen.findByText("Draft Detour 123"))
+    await userEvent.click(deleteDetourButton.get())
+
+    expect(
+      await screen.findByText("Are you sure you want to delete this draft?")
+    ).toBeVisible()
+  })
+
+  test("Clicking the Delete Draft button on the Delete Draft Detour confirmation modal calls the deleteDetour api function", async () => {
+    jest
+      .mocked(fetchDetours)
+      .mockResolvedValue(Ok(detourListFactoryWithDraft.build()))
+    jest.mocked(fetchDetour).mockResolvedValue(Ok(draftDetourFactory.build()))
+
+    render(<DetourListPage />)
+
+    await userEvent.click(await screen.findByText("Draft Detour 123"))
+    await userEvent.click(deleteDetourButton.get())
+
+    expect(
+      await screen.findByText("Are you sure you want to delete this draft?")
+    ).toBeVisible()
+
+    await userEvent.click(confirmDeleteDetourButton.get())
+
+    expect(deleteDetour).toHaveBeenCalledWith(123)
+  })
+
+  test("Clicking the Cancel button on the Delete Draft Detour confirmation modal closes the modal", async () => {
+    jest
+      .mocked(fetchDetours)
+      .mockResolvedValue(Ok(detourListFactoryWithDraft.build()))
+    jest.mocked(fetchDetour).mockResolvedValue(Ok(draftDetourFactory.build()))
+
+    render(<DetourListPage />)
+
+    await userEvent.click(await screen.findByText("Draft Detour 123"))
+    await userEvent.click(deleteDetourButton.get())
+
+    expect(
+      await screen.findByText("Are you sure you want to delete this draft?")
+    ).toBeVisible()
+
+    await userEvent.click(cancelButton.get())
+
+    expect(
+      await screen.queryAllByText("Are you sure you want to delete this draft?")
+    ).toStrictEqual([])
+  })
```

Comment:
Can we use regular `queryByText`? and the RTL library provides us with `not.toBeInTheDocument`, which works with the result of `queryBy` IIRC.

And I don't _think_ `queryBy` functions return a promise.
```suggestion
    expect(
      screen.queryByText("Are you sure you want to delete this draft?")
    ).not.toBeInTheDocument()
  })
```